packages/phoenix-evals/src/phoenix/evals/tracing.py

```diff
@@ -0,0 +1,195 @@
+            f"{host_root_path}. Value cannot end with '/'"
+        )
+    return host_root_path
+
+
+def get_env_client_headers() -> dict[str, str]:
+    headers = parse_env_headers(getenv(ENV_PHOENIX_CLIENT_HEADERS))
+    if (api_key := get_env_phoenix_api_key()) and "authorization" not in [
+        k.lower() for k in headers
+    ]:
+        headers["Authorization"] = f"Bearer {api_key}"
+    return headers
+
+
+def get_env_collector_endpoint() -> Optional[str]:
+    return getenv(ENV_PHOENIX_COLLECTOR_ENDPOINT) or getenv(ENV_OTEL_EXPORTER_OTLP_ENDPOINT)
+
+
+def get_base_url() -> str:
+    host: str = get_env_host()
+    if host == "0.0.0.0":
+        host = "127.0.0.1"
+    base_url: str = get_env_collector_endpoint() or f"http://{host}:{get_env_port()}"
+    return base_url
+
+
+@overload
+def getenv(key: str) -> Optional[str]: ...
+@overload
+def getenv(key: str, default: str) -> str: ...
+def getenv(key: str, default: Optional[str] = None) -> Optional[str]:
+    """
+    Retrieves the value of an environment variable.
+
+    Parameters
+    ----------
+        key : str
+            The name of the environment variable.
+        default : Optional[str], optional
+            The default value to return if the environment variable is not set, by default None.
+
+    Returns
+    -------
+    Optional[str]
+        The value of the environment variable, or `default` if the variable is not set.
+        Leading and trailing whitespaces are stripped from the value, assuming they were
+        inadvertently added.
+    """
+    if (value := os.getenv(key)) is None:
+        return default
+    return value.strip()
+
+
+def get_tracer() -> Tracer:
+    """
+    1. First check if there's a global tracer provider and pull from that
+    2. If not, create its own tracer using environment variables to configure the span processor
+    3. If there's any failure, return a no-op processor
+    """
+    try:
+        global_tracer_provider = trace_api.get_tracer_provider()
+
+        if (
+            hasattr(global_tracer_provider, "get_tracer")
+            and not global_tracer_provider.__class__.__name__ == "NoOpTracerProvider"
+        ):
+            tracer = global_tracer_provider.get_tracer(__name__)
+            resource = getattr(global_tracer_provider, "_resource", Resource({}))
+            return tracer
+    except Exception:
+        logger.debug("No global tracer provider")
+
+    try:
+        resource = Resource({})
+        tracer_provider = trace_sdk.TracerProvider(resource=resource)
+
+        base_url = get_base_url()
+        endpoint = urljoin(base_url, "v1/traces")
+        span_processor: trace_sdk.SpanProcessor = SimpleSpanProcessor(
+            OTLPSpanExporter(
+                endpoint=endpoint,
+                headers=get_env_client_headers(),
+            )
+        )
+        tracer_provider.add_span_processor(span_processor)
+        return tracer_provider.get_tracer(__name__)
```

Comment:
I would nix this. The library should be natively instrumented but not setup the exporter. The tracing needs to work irregareless of tracing backends.